### PR TITLE
Added tests for Set/Delete Immutability Policy and Set Legal Hold on Blob Snapshots and Versions

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/autorest.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Run `dotnet build /t:GenerateCode` to generate code.
 
 ``` yaml
 input-file:
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/f6f50c6388fd5836fa142384641b8353a99874ef/specification/storage/data-plane/Microsoft.BlobStorage/stable/2024-08-04/blob.json
+    - C:\azure-rest-api-specs\specification\storage\data-plane\Microsoft.BlobStorage\stable\2025-01-05\blob.json
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/src/autorest.md:7]: TODO fix this when https://github.com/Azure/azure-rest-api-specs/pull/30649 merges

**Review by vincenttran-msft — APPROVED** [review]
🚢

File: sdk/storage/Azure.Storage.Blobs/tests/ImmutableStorageWithVersioningTests.cs [hunk 2]
```diff
@@ -562,6 +563,92 @@ public async Task DeleteImmutibilityPolicyAsync()
...
+
+                // Act
+                await snapshotClient.SetImmutabilityPolicyAsync(immutabilityPolicy);
+
+                // Assert that the base blob does not have an immutability policy.
+                Response<BlobProperties> propertiesResponse = await blob.GetPropertiesAsync();
+                Assert.IsNull(propertiesResponse.Value.ImmutabilityPolicy.ExpiresOn);
+                Assert.IsNull(propertiesResponse.Value.ImmutabilityPolicy.PolicyMode);
+
+                // Asser that the blob snapshot has an immuability policy.
```
Comment (vincenttran-msft) [inline sdk/storage/Azure.Storage.Blobs/tests/ImmutableStorageWithVersioningTests.cs:591]: supernit: missing 't' in "Assert"

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/tests/ImmutableStorageWithVersioningTests.cs]: Fixed!

**Pull request merged**